python/ray/tests/conftest.py

```diff
@@ -361,7 +361,8 @@ def start_cluster(ray_start_cluster_enabled, request):
     assert request.param in {"ray_client", "no_ray_client"}
     use_ray_client: bool = request.param == "ray_client"
     cluster = ray_start_cluster_enabled
-    cluster.add_node(num_cpus=4)
+    # Use a hard-coded agent port to prevent failures due to port conflicts.
+    cluster.add_node(num_cpus=4, metrics_agent_port="10031")
```

Comment:
To prevent regression, is there any easy way to `assert` whether the port is already in use? I assume not since you would somehow need to check this inside the cluster?

Reply:
Good point, I'm not sure how to do this... if it were straightforward we would probably just be checking that inside Ray to prevent the port conflict always.  It might be easy though, @rkooo567 any ideas about this?